### PR TITLE
Create Movies Page

### LINKED ISSUE #9: Movies Page
As an authenticated user,
When I visit the movies page,
I should see the 40 results from my search,
I should also see the "Find Top Rated Movies" button and the Find Movies form at the top of the page.

**Details:** The results from the search should appear on this page, and there should only be a maximum of 40 results. The following details should be listed for each movie.
- [x] Title (As a Link to the Movie Details page)
- [x] Vote Average of the movie

### PR DESCRIPTION
**Pull request by ETBassist**
- Create test for movie page top rated 40
- Create top 40 movies view with API call
- Added test to check search field and scaffold out future tests
- Refactor controller, changes nothing
- Update text field tag to search tag and add sections
- Add encrypted moviedb api key to travis file
- Unskip tests and comment out within block for search results
- Add method: :get to search forms
- Add search functionality and some sad path handling to movies index action
- Remove page parameter for first search api call
- Move no movies found message into view and uncomment within block in spec
- Change movie titles to links on movies#index page

_X___ Wrote Tests _X___ Implemented ____ Reviewed

Neccesary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes:

    description closes #9

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link: Coyote Searching

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  def conn
+    Faraday.new(url: 'https://api.themoviedb.org') do |faraday|
+      faraday.params['api_key'] = ENV['MOVIE_DB_API_KEY']
+    end
+  end
+
+  def get_json(url)
+    response = conn.get(url)
+    JSON.parse(response.body, symbolize_names: true)
+  end
```
Comment (helloeduardo) [inline app/controllers/movies_controller.rb:28]: Nice job extracting this into private methods for this iteration

File: app/views/movies/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  <%= form_tag('/movies', method: :get) do %>
+    <%= search_field_tag :search, nil, placeholder: 'Search By Movie Title' %>
+    <%= submit_tag 'Search For Movies' %>
+  <% end %>
+</section>
+
+<section id="results">
+  <% if @movies.empty? %>
+    <p>No movies have matched your request! Please try again!</p>
+  <% else %>
```
Comment (helloeduardo) [inline app/views/movies/index.html.erb:13]: Thanks for adding these sections and conditionals!

File: spec/features/movies/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      fill_in :search, with: 'Love'
+      click_on 'Search For Movies'
+      expect(current_path).to eq('/movies')
+
+
+      # How can we future proof this test to not break later on?
+      within '#results' do
+        expect(page).to have_content("Love")
+        expect(page).to have_content("Vote Average: ")
+      end
```
Comment (helloeduardo) [inline spec/features/movies/index_spec.rb:37]: Good place to use a VCR cassette perhaps?

File: spec/features/movies/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    # Here we just want the search to fail for sad path
+    it 'If no movie matches search result, user will recieve a message' do
+      fill_in :search, with: 'The balloon popped and it was gibberish'
+      click_on 'Search For Movies'
+      expect(current_path).to eq('/movies')
+
+      within '#results' do
+        expect(page).to have_content("No movies have matched your request! Please try again!")
+      end
```
Comment (helloeduardo) [inline spec/features/movies/index_spec.rb:49]: Love the example search for the sad path scenario! 😂

**Review by helloeduardo — APPROVED** [review]
Thanks for adding all of these improvements!

File: spec/features/movies/index_spec.rb (hunk 3 above)
Comment (ETBassist) [reply spec/features/movies/index_spec.rb:37]: Haha, I was just thinking the same thing

**Pull request merged**